### PR TITLE
add subworkflow bam_markduplicates_picard

### LINKED ISSUE #2168: new subworkflow bam_markduplicates_picard

### PR DESCRIPTION
**Pull request by dmarron**
## PR checklist

Closes #2168 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/bam_markduplicates_picard/meta.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - idxstats:
+      type: file
+      description: File containing samtools idxstats output
+      pattern: "*.{idxstats}"
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+authors:
+  - "@dmarron"
```
Comment (Emiller88) [inline subworkflows/nf-core/bam_markduplicates_picard/meta.yml:60]:
```suggestion
  - "@dmarron"
  - "@drpatelh"
```

File: tests/subworkflows/nf-core/bam_markduplicates_picard/main.nf [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+workflow test_bam_markduplicates_picard {
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
+    ]
+    fasta = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+
+    BAM_MARKDUPLICATES_PICARD ( input, fasta )
+}
```
Comment (Emiller88) [inline tests/subworkflows/nf-core/bam_markduplicates_picard/main.nf:15]: Can we test cram input as well?

**Review by Emiller88 — CHANGES_REQUESTED** [review]
Just need a cram test to make sure those are working as expected!

**Review by Emiller88 — APPROVED** [review]
Awesome work! Good to merge if the tests pass!

File: subworkflows/nf-core/bam_markduplicates_picard/main.nf [hunk 3]
```diff
@@ -0,0 +1,52 @@
+//
+// Picard MarkDuplicates, index BAM file and run samtools stats, flagstat and idxstats
+//
+
+include { PICARD_MARKDUPLICATES      } from '../../../modules/nf-core/picard/markduplicates/main'
```
Comment (drpatelh) [inline subworkflows/nf-core/bam_markduplicates_picard/main.nf:5]:
```suggestion
include { PICARD_MARKDUPLICATES } from '../../../modules/nf-core/picard/markduplicates/main'
```

**Review by drpatelh — APPROVED** [review]
🚀

**Pull request merged**